### PR TITLE
feat:produce sine and saw tooth waves through wave generator

### LINKED ISSUE #1012: Producing sine triangular waveforms through wave generator module
**Actual Behaviour**

Wave generator UI is working but it needs to be connected to the communication module so that it can connect to the PSLab device.

**Expected Behaviour**

The waveform panel should be connected to the communication module and produce sine waves.

**Steps to reproduce it**

Wave Generator for waveform panel 

Science Lab for communication functions.

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by Avjeet**
Fixes #1012 

**Changes**:Connected the Wavegenerator module to the Science Lab module to produce sine wave and saw tooth wave.

**Screenshot/s for the changes**: 
ezgif com-video-to-gif 4

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x ] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [x ] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [ ] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [ ] My code does not contain any extra lines or extra spaces
- [ ] I have requested reviews from other members

**APK for testing**: Currently the work is in progress

### CODE REVIEW

**Comment by Avjeet** [comment]
@CloudyPadmal plz add the work in progress tag

File: app/src/main/java/org/fossasia/pslab/activity/WaveGeneratorActivity.java [hunk 1]
```diff
@@ -63,80 +105,107 @@
...
-    private static String btn_Active = "";
+
+    private static String waveBtnActive = "";
+    private static String pwmBtnActive = "";
     private static String prop_active = "";
+    private static boolean waveMonSelected;
+    private TextView activePropTv = null;
+    private TextView activePwmPinTv = null;
 
+    ScienceLab scienceLab ;
```
Comment (Vikum94) [inline app/src/main/java/org/fossasia/pslab/activity/WaveGeneratorActivity.java]: You can make this private

File: app/src/main/java/org/fossasia/pslab/activity/WaveGeneratorActivity.java [hunk 2]
```diff
@@ -197,44 +383,137 @@ public void onStopTrackingTouch(IndicatorSeekBar seekBar) {
...
+                pwmDutyValue.setText(valueText);
 
-        Double value = (double) WaveGeneratorCommon.wave.get(btn_Active).get(PHASE);
-        String valueText = value + " " + getString(R.string.deg_text);
-        wavePhaseValue.setText(valueText);
+                value = (double) WaveGeneratorCommon.wave.get(pwmBtnActive).get(PHASE);
+                valueText = value + " " + getString(R.string.deg_text);
+                pwmPhaseValue.setText(valueText);
+                break;
+        }
```
Comment (Vikum94) [inline app/src/main/java/org/fossasia/pslab/activity/WaveGeneratorActivity.java]: Add default tag here for the switch statement

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -807,4 +809,4 @@
     \u2022 Then go to the Configure tab from the bottom navigation bar and choose the BH-1750 in the drop down
     menu Select Sensor.\n</string>
 
-</resources>
+</resources>
```
Comment (CloudyPadmal) [inline app/src/main/res/values/strings.xml]: Please don't remove these extra lines auto generated by IDE

File: app/src/main/java/org/fossasia/pslab/activity/OscilloscopeActivity.java [hunk 4]
```diff
@@ -252,6 +252,9 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
             case "Speed of Sound":
                 isSpeedOfSoundExperiment = true;
                 break;
+            case "WaveGenerator":
+                //do nothing
+                break;
```
Comment (CloudyPadmal) [inline app/src/main/java/org/fossasia/pslab/activity/OscilloscopeActivity.java]: What's this one doing here?

Comment (Avjeet) [reply app/src/main/java/org/fossasia/pslab/activity/OscilloscopeActivity.java]: without this the oscilloscope is crashing

**Pull request merged**